src/ui/dialogs/contactsearch.ts

```diff
@@ -0,0 +1,253 @@
+import Dialog from '../Dialog'
+import Contact from '../../Contact'
+import Form from '../../connection/Form'
+import Log from '../../util/Log'
+import { IConnection } from '@connection/Connection.interface';
+import Translation from '@util/Translation';
+import Client from '../../Client'
+import Account from '../../Account'
+import JID from '../../JID'
+
+let dialog: Dialog;
+
+export const CANCELED = 'canceled';
+
+let contactSearchTemplate = require('../../../template/contactsearch.hbs');
+
+export default function(username?: string) {
+
+    let content = contactSearchTemplate({
+      accounts: Client.getAccountManager().getAccounts().map(account => ({
+         uid: account.getUid(),
+         jid: account.getJID().bare,
+      }))
+    });
+
+    dialog = new Dialog(content);
+    dialog.open();
+
+    let accountId = <string> $('#jsxc-account').val();
+
+    let account = Client.getAccountManager().getAccount(accountId);
+
+    let connection = account.getConnection();
+
+    getSearchServiceDomain(connection,account).then((services) => {
+        if (services&&services.length>0)
+        {
+            connection.getSearchService().getSearchForm(services[0].domain)
+              .then(stanza => Form.fromXML(stanza))
+              .then((form: Form) => {
+                 return showForm(form, connection,services[0]);
+            }).then(function(searchresult :Element ){ showResult(searchresult,account);});
+        }
+        else
+        {
+            dialog.close();
+            alert(Translation.t('not_available'));
+        }
+    });
+
+}
+
+function getSearchServiceDomain(connection : IConnection, account : Account)
```

Comment:
Should be converted to an async function.

Reply:
done